src/core/server/http/integration_tests/request.test.ts

```diff
@@ -197,4 +197,24 @@ describe('KibanaRequest', () => {
       });
     });
   });
+
+  describe('request id', () => {
+    it('accepts x-opaque-id header case-insensitively', async () => {
+      const { server: innerServer, createRouter } = await server.setup(setupDeps);
+      const router = createRouter('/');
+      router.get({ path: '/', validate: false }, async (context, req, res) => {
+        return res.ok({ body: { requestId: req.requestId } });
+      });
+      await server.start();
+
+      const st = supertest(innerServer.listener);
+
+      const resp1 = await st.get('/').set({ 'x-opaque-id': 'alpha' }).expect(200);
+      expect(resp1.body).toEqual({ requestId: 'alpha' });
+      const resp2 = await st.get('/').set({ 'X-Opaque-Id': 'beta' }).expect(200);
+      expect(resp2.body).toEqual({ requestId: 'beta' });
+      const resp3 = await st.get('/').set({ 'X-OPAQUE-ID': 'gamma' }).expect(200);
+      expect(resp3.body).toEqual({ requestId: 'gamma' });
```

Comment:
When is this case conversion done? Couldn't find it in `src/core/server/http/router/request.ts`

Reply:
If I remember correctly, Hapi does this. I figured it would be good to include this test to make removing Hapi easier.